submit, bug fixes, dashboard messages

panjenny0 opened the pull request:
### Dashboard message
- updated RSVP-by date to Jan 3rd 2021 (FYI @JPGarCar)
- updated copy text wording and embedded links
- fetching dates from firebase in `InternalWebsites/Livesite`: `sendAcceptancesBy`, `rsvpBy`, `offWaitlistNotify`

### [bugfix] Redirect to `/application/part-1` when application is not submitted
- added condition for `/application` in the `useEffect` of `Auth.js` to check if applicant has submitted their application before being able to access the dashboard - if their application is not submitted, redirect to `/application/part-1` (FYI @andykwan1124)
```
if (location === '/application') {
        await handleUser(setUser, setLocation)
      }
```
- I noticed that the user gets automatically redirected to `/submission` if they RSVP and refresh the page (or navigate away and come back) - I've restricted the redirect to `/submission` for HackCamp only

### [bugfix] User document fields get deleted when you click 'Back' several times in `/application`
- added `!applicationRef.current` in the first check in `forceSave` (thank you @jackyzha0 so big brained 🧠)
```
const forceSave = useCallback(async () => {
    if (!user || !applicationRef.current) return
    const updatedApp = {
      ...applicationRef.current,
      submission: {
        lastUpdated: firebase.firestore.Timestamp.now(),
        submitted: false,
      },
    }
```

### Other minor changes
- changed review page to display `otherEngagementSource` if `Other` is chosen for `engagementSource`
- changed width and border radius of review container to align with design specs
- also fixes bugs listed here https://github.com/nwplus/livesite/pull/220#issuecomment-743953305

### How to test

#### testing dashboard redirect
1. go to https://nwplus-ubc--pr220-date-fix-yjvu589n.web.app/application/part-1 and login
2. go to `/application` - you should be redirected to part 1 of the app
note that after you click on 'submit' on the application, you should no longer get redirected

#### testing dashboard messages 
1. assuming you've logged in, access your Applicant document in the database: https://console.firebase.google.com/u/0/project/nwplus-ubc-dev/firestore/data~2FHackathons~2FnwHacks2021~2FApplicants
2. change your `applicantStatus` to `waitlisted` then go back to `/application` in the preview link https://nwplus-ubc--pr220-date-fix-yjvu589n.web.app/application and refresh - the status and message should update to waitlisted
3. change your `applicantStatus` to `accepted` then go back to preview link and refresh - message should update

kevin-zou reviewed (approved):
lgtm! 💯

vincentchiang1 reviewed (dismissed):
💯

anneguo3 commented on src/components/ApplicationForm/ReviewCards.js:
```diff
@@ -283,8 +294,26 @@ export default ({ formInputs, handleEdit, onChange }) => {
+              I authorize you to share my application/registration information for event
+              administration, ranking, MLH administration, pre- and post-event informational
+              e-mails, and occasional messages about hackathons in-line with the{' '}
+              <A bolded color="primary" href="https://mlh.io/privacy" target="_blank">
```
omg just wanted to say than u for adding the links ❤️

panjenny0 commented:
Addressed some issues/bugs brought up by @anneguo3 (thank u!)
- [x] upon submission, go back to main page of portal (should not be able to see the pages anymore)
<img width="530" alt="Screen Shot 2020-12-12 at 9 21 22 PM" src="https://user-images.githubusercontent.com/38872354/102003860-00d0bf80-3cc0-11eb-8d7f-bdeee5b727c6.png">

- [x] Back on first page —> go “back to portal”
<img width="1029" alt="Screen Shot 2020-12-12 at 8 59 00 PM" src="https://user-images.githubusercontent.com/38872354/102003529-696a6d00-3cbd-11eb-8dd8-d1c12edb5013.png">
_(I made this 'Save & Logout')_

- [x] limit infinite scroll on last page 
<img width="1680" alt="Screen Shot 2020-12-12 at 9 03 28 PM" src="https://user-images.githubusercontent.com/38872354/102003545-7edf9700-3cbd-11eb-9614-22a3dc140d44.png">
_(lol this is the best I could do)_
- [x] add stars to the check boxes
- [x] default grad, number of hackathons as empty, and isOfLegalAge as null (unselected) 
<img width="699" alt="Screen Shot 2020-12-12 at 8 20 43 PM" src="https://user-images.githubusercontent.com/38872354/102003838-d54dd500-3cbf-11eb-89b1-cb23d3f5e8b7.png">
<img width="992" alt="Screen Shot 2020-12-12 at 8 20 16 PM" src="https://user-images.githubusercontent.com/38872354/102003841-d7b02f00-3cbf-11eb-90fc-67f248f93eea.png">
_(to test this, delete your document in the db; the new changes should initialize null as initial values)_

jackyzha0 commented on src/utility/HackerApplicationContext.js:
```diff
@@ -30,7 +30,7 @@ export function HackerApplicationProvider({ children }) {
   /**Uses a reference to the application because I don't want all my useEffects triggering every time someone changes the application. */
   const forceSave = useCallback(async () => {
-    if (!user) return
+    if (!user || !applicationRef.current) return
```
bless up

jackyzha0 commented on src/utility/Constants.js:
```diff
@@ -101,6 +101,7 @@ export const hackerApplicationTemplate = Object.freeze({
 // relevant dates for each status message displayed on the dashboard
 export const relevantDates = Object.freeze({
   sendAcceptancesBy: 'December 30th, 2020',
-  rsvpBy: 'Janurary 7th at 11:59 PM (Pacific Time)',
+  rsvpBy: 'Janurary 3rd at 11:59 PM (Pacific Time)',
+  offWaitlist: 'one week before the event',
   hackathonDate: 'January 9-10th',
 })
```
is there a reason these aren't on firebase? livesite has some fields related to these

panjenny0 replied:
Oh I totally didn't realize that there were similar fields on firebase
We were just a little short on time and I needed to make this change really quickly 😢 but let me just manually add some new fields and reference from firebase instead

panjenny0 replied:
Ok updated! Pulling from firebase now

andykwan1124 reviewed (approved):
LGTM! nice fixes!!!!!
